Lib/ipaddress.py

```diff
@@ -1092,20 +1084,41 @@ def _make_netmask(cls, arg):
         Argument can be:
         - an integer (the prefix length)
         - a string representing the prefix length (e.g. "24")
-        - a string representing the prefix netmask (e.g. "255.255.255.0")
+        - a string representing the netmask or hostmask (e.g. "255.255.255.0")
+        - a bytes object representing a netmask or hostmask
+          (e.g. b"\xff\xff\xff\x00")
+        - an IPv4Address object representing the netmask or hostmask
```

Comment:
@MoritzS Why this method should support these types?
I prefer previous code.

FYI, since bytes and str are supported now, `BytesWarning` can be raised
while ` if arg not in cls._netmask_cache`.